test/testsuites/clusteroperatorstatustests.go

```diff
@@ -0,0 +1,70 @@
+package testsuites
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// ClusterOperatorStatusOnStartup is a test suite that ensures the ClusterOperator resource which
+// defines the status of the marketplace operator has the correct status upon initialization
+func ClusterOperatorStatusOnStartup(t *testing.T) {
+	ctx := test.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	// Get global framework variables.
+	client := test.Global.Client
+
+	// Get namespace
+	namespace, err := test.NewTestCtx(t).GetNamespace()
+	require.NoError(t, err, "Could not get namespace")
+
+	// Check that the ClusterOperator resource has the correct status
+	clusterOperatorName := "marketplace"
+	expectedTypeStatus := map[configv1.ClusterStatusConditionType]configv1.ConditionStatus{
+		configv1.OperatorProgressing: configv1.ConditionFalse,
+		configv1.OperatorAvailable:   configv1.ConditionTrue,
+		configv1.OperatorDegraded:    configv1.ConditionFalse}
+	err = clusterOperatorCheckTypeStatus(client, clusterOperatorName, namespace, expectedTypeStatus)
+	require.NoError(t, err, "ClusterOperator never reached expected status")
+}
+
+// clusterOperatorCheckTypeStatus checks each condition (type & status) against expectedTypeStatus
+func clusterOperatorCheckTypeStatus(client test.FrameworkClient, clusterOperatorName string, namespace string, expectedTypeStatus map[configv1.ClusterStatusConditionType]configv1.ConditionStatus) error {
+	// Get ClusterOperator resource with clusterOperatorName name
+	namespacedName := types.NamespacedName{Name: clusterOperatorName, Namespace: namespace}
+	result := &configv1.ClusterOperator{}
+
+	// Poll in case the ClusterOperator resource is not immediately present on the cluster
+	RetryInterval := time.Second * 5
+	Timeout := time.Minute * 5
+	err := wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
```

Comment:
Why can't you just use the helper method here?

`helpers.WaitForResult(client, result, clusterOperatorName, namespace)`